packages/core/src/engine/Autodiff.ts

```diff
@@ -1125,857 +1127,861 @@ export const fns = {
+            stmt = `const ${parName} = ${childName0} / (${childName1} + ${EPS_DENOM});`;
+        } else if (z.op === "atan2") {
+            stmt = `const ${parName} = Math.atan2(${childName0}, ${childName1});`;
+        } else {
+            stmt = `const ${parName} = ${childName0} ${op} ${childName1};`;
+        }
+
+        stmts.push(stmt);
+
+        if (z.debug) {
+            const stmt2 = `console.log("${z.debugInfo} (var ${parName}) | value: ", ${parName}, "during ${setting} evaluation");`;
+            stmts.push(stmt2);
+        }
+
+        // Array efficiency?
+        return {
+            counter: parCounter,
+            prog: res0.prog.concat(res1.prog).concat(stmts),
+            inputs: res0.inputs.concat(res1.inputs),
+            output: parName,
+            references: [],
+        };
+    } else {
+        // N-ary node
+        const childNodes = z[childType].map((e) => e.node);
+
+        const childNames = [];
+        let prog: string[] = [];
+        let inputs: string[] = [];
+        let counter = i;
+
+        // Evaluate each child and get its generated code, inputs, and name first
+        for (const childNode of childNodes) {
+            const res = traverseGraph(counter, childNode, setting);
+            prog = prog.concat(res.prog);
+            inputs = inputs.concat(res.inputs);
+
+            // Child was already visited; don't generate code again, just a reference
+            if (res.references[0]) {
+                childNames.push(res.references[0]);
+                counter = res.counter;
+            } else {
+                childNames.push(c + String(res.counter));
+                counter = res.counter + 1;
+            }
+        }
+
+        const parName = c + String(counter);
+
+        // Mark node as visited with name as reference
+        // TODO: factor out these 2 lines from all cases
+        z.nodeVisited = true;
+        z.name = parName;
+
+        const op = z.op;
+        const stmts = [];
+        let stmt;
+
+        // Deals with ifCond nodes (ternary)
+        // (eval c; eval d; eval e; const xNUM = c ? d : e;)
+        // codegen doesn't short-circuit -- it generates code for both branches of the `if` first
+        // TODO: Fix this so it doesn't *evaluate* both branches of the `if`, just the one that's relevant to the condition
+
+        if (op === "ifCond") {
+            if (childNames.length !== 3) {
+                log.trace("args", childNames);
+                throw Error("expected three args to if cond");
+            }
+
+            stmt = `const ${parName} = ${childNames[0]} ? ${childNames[1]} : ${childNames[2]};`;
+        } else if (op === "+ list") {
+            const childList = "[".concat(childNames.join(", ")).concat("]");
+            // TODO: revert the console.log. also probably name the child list?
+            stmt = `console.log("reduce", ${childList}.length);\n
+            const ${parName} = ${childList}.reduce((x, y) => x + y); `;
```

Comment:
fwiw replacing this code with the loop version avoids allocating the closure, thus avoiding the stack blow-up

```suggestion
            let ${parName} = 0;\n
            for (const $child of ${childList}) {\n
              ${parName} += $child;\n
            }`;
```

Reply:
Amazing, thanks! I'll try it out. This is a much simpler solution than the one I thought of originally.

Reply:
yeah I tried it locally and it seems to work (as long as parName isn't `$child`)